src/modules/mim/firewall.json

```diff
@@ -1,41 +1,305 @@
+            "type": "enum",
+            "valueSchema": "integer",
+            "enumValues": [
+              {
+                "name": "unknown",
+                "enumValue": 0
+              },
+              {
+                "name": "enabled",
+                "enumValue": 1
+              },
+              {
+                "name": "disabled",
+                "enumValue": 2
+              }
+            ]
+          }
         },
-        { 
-          "name": "firewallFingerprint", 
-          "type": "mimObject", 
+        {
+          "name": "firewallFingerprint",
+          "type": "mimObject",
           "desired": false,
           "schema": "string"
+        },
+        {
+          "name": "firewallRule",
+          "type": "mimObject",
+          "desired": true,
+          "schema": {
+            "type": "object",
+            "fields": [
+              {
+                "name": "action",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "add",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "delete",
+                      "enumValue": 2
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "command",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "accept",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "reject",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "drop",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "type",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "inbound",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "outbound",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "forward",
+                      "enumValue": 3
+                    }
+                  ]
+                }
```

Comment:
Can replace this enum with a boolean "outbound" (true out, false in, FW does not really matter)

Reply:
Changed to inbound (default true) since this is the more common rule type